scripts/cargo-rmc

```diff
@@ -94,6 +63,134 @@ def main():
     else:
         return rmc.run_cbmc(cbmc_filename, args.cbmc_args, args.verbose, args.quiet, args.dry_run)
 
+def parse_args():
+    # Create parser
+    def create_parser():
+        parser = argparse.ArgumentParser(prog="cargo rmc", description="Verify a Rust crate. For more information, see https://github.com/model-checking/rmc.")
+
+        crate_group = parser.add_argument_group("Crate", "You can pass in the rust crate positionally or with the --crate flag.")
+        crate_group.add_argument("crate", help="crate to verify", nargs="?")
+        crate_group.add_argument("--crate", help="crate to verify", dest="crate_flag", metavar="CRATE")
+
+        config_group = parser.add_argument_group("Config", "You can configure cargo rmc with your Cargo.toml file.")
+        config_group.add_argument("--config-toml", help="Where to read configuration from; defaults to crate's Cargo.toml")
+        config_group.add_argument("--no-config-toml", action="store_true", help="Do not use any configuration toml")
+
+        exclude_flags = [
+            # This should be able to be supported; https://github.com/model-checking/rmc/issues/359
+            "--gen-symbols",
+            # This should be able to be supported; https://github.com/model-checking/rmc/issues/360
+            "--allow-cbmc-verification-failure",
+        ]
+        rmc_flags.add_flags(parser, {"default-target": "target"}, exclude_flags=exclude_flags)
+        
+        return parser
+
+    # Check for conflicting flags
+    def validate(args):
+        rmc.ensure(not (args.crate and args.crate_flag), "Please provide a single crate to verify.")   
+        rmc.ensure(not (args.no_config_toml and args.config_toml), "Incompatible flags: --config-toml, --no-config-toml")
+
+    # Fix up args before returning
+    def post_process(args):
+        # Combine positional and flag argument for input
+        cl_args.crate = cl_args.crate or cl_args.crate_flag or "."
+
+        # --quiet overrides --verbose
+        if args.quiet:
+            args.verbose = False
+
+        # Add some CBMC flags by default unless `--no-default-checks` is being used
+        if args.default_checks:
+            rmc.add_selected_default_cbmc_flags(args)
+
+    # Combine args set by config toml and provided on command line
+    def get_combined_args(config_toml, crate, cl_args):
```

Comment:
The logic of this function was hard to follow

Reply:
I've tried to rearrange the comments a bit, but I don't know how to decompose the function much more. I could break it into helpers but I'm worried that will just create a flood of helper functions.

Reply:
Proposed logic:
```python
if no_config_toml:
    return cl_args
elif config_toml is not None:
    # If config_toml is explicitly given, ensure it exists
    rmc.ensure(pathlib.Path(config_toml).is_file(), f"Invalid path to config file: {config_toml}")
else:
    crate_cargo_toml = pathlib.Path(crate).joinpath("Cargo.toml")
        rmc.ensure(pathlib.Path(config_toml).is_file(), f"Cannot find configuration toml at expected location: {config_toml}")
```

Reply:
One other thought: make this function just take a toml, and have it do the combining. then have a seperate logic to determine what cargo toml to use.

Reply:
The logic is now to use `get_config_toml` to find the config toml file (or `None`), and then when the result is present, we use `get_combined_args` to actually parse the config toml and combine it with the command line args.